backend/src/v2/metadata/client.go

```diff
@@ -120,7 +120,9 @@ func NewClient(serverAddress, serverPort string, tlsEnabled bool) (*Client, erro
 
 	creds := insecure.NewCredentials()
 	if tlsEnabled {
-		config := &tls.Config{}
+		config := &tls.Config{
+			InsecureSkipVerify: true, // This should be removed by https://issues.redhat.com/browse/RHOAIENG-13871
```

Comment:
can you elaborate on why this is needed? It effectively makes TLS pointless.

Reply:
TLS for MLMD gRPC server is not fully implemented while https://issues.redhat.com/browse/RHOAIENG-13871 is not resolved.
At this point, the client doesn't validate the server's certificate.

Reply:
[My previous PR that adds support for TLS to MLMD](https://github.com/opendatahub-io/data-science-pipelines/commit/5586f80f95166a06ffcf4849750f190492215e2d) doesn't add any validation on the client side. 
Does that make TLS pointless? Yes. We noticed this after merging the PR and then we created https://issues.redhat.com/browse/RHOAIENG-13871 to fix it.

In the meantime, we noticed that the importer feature was broken by [my previous PR that adds support for TLS to MLMD](https://github.com/opendatahub-io/data-science-pipelines/commit/5586f80f95166a06ffcf4849750f190492215e2d). That's what this PR aims to solve.

Having `InsecureSkipVerify: true` is completely different than `PodToPodTLS: false`.
While `InsecureSkipVerify: true` is in the MLMD gRPC server scope, `PodToPodTLS: false` is in the entire DSPA scope.

That said, this PR is not removing anything that existed before. OK, so why to add `InsecureSkipVerify: true` now? Because without that, the client complains about the server's certificate not being signed by a known CA. For some reason I don't know, that didn't happen on my first PR.